### DISCUSSION TITLE
i'm now adding transmit controles to android client, but i have a question to continue

### DISCUSSION THREAD

**Discussion by peter1384** [BearWare/TeamTalk5#2247] [Q&A]
### Description

hello @bear101 When i was adding intercept subscriptions to the android client, I was aware what to call for its function so to request that from server, and they was These.
ttclient.doSubscribe(user.nUserID, Subscription.SUBSCRIBE_INTERCEPT_EXAMPLE);
This is for subscribing and
ttclient.doUnsubscribe(user.nUserID, Subscription.SUBSCRIBE_INTERCEPT_EXAMPLE);
This is for unsubscribing
Now, I don't know What should i call for these lines for transmit controles, Can you please send one of these lines for me as an example for me in comment? thanks.

### Application

- [ ] qtTeamTalk
- [X] TeamTalkAndroid
- [ ] iTeamTalk
- [ ] TeamTalkServer

**Comment by bear101** [comment]
For controlling transmission in channels you can maybe look into this test-case:
https://github.com/BearWare/TeamTalk5/blob/master/Library/TeamTalkJNI/test/dk/bearware/TeamTalkServerTestCase.java#L1241

**Comment by peter1384** [comment]
@bear101 i've looked at this, but i stil couldn't find out what should i call after ttclient.do for turning off and turning on a transmit for a user. Note that i'm not implementing classroom transmits now, i'm just doing normal transmissions which are done by ctrl+alt+q throo t in windows clients.
So, can you please show that to me? Thanks.

**Reply by bear101** [reply]
Channel.transmitUsers is a two dimensional array. Column (y-axis) has index 0 as user id and streamtype as index 1.
```
chan.transmitUsers[0] = 57;
chan.transmitUsers[1] = StreamType.STREAMTYPE_VOICE;
ttclient.doUpdateChannel(chan);
```
With the above code user ID 57 can transmit voice in a ChannelType.CHANNEL_CLASSROOM channel.

To reset:
```
chan.transmitUsers[0] = 57;
chan.transmitUsers[1] = StreamType.STREAMTYPE_NONE;
ttclient.doUpdateChannel(chan);
```

**Comment by peter1384** [comment]
@bear101 ok, This now makes sence. Now tell me what about a non classroom channel?

**Reply by bear101** [reply]
For non-classroom the chan.transmitUsers has the opposite effect

**Comment by peter1384** [comment]
@bear101 i didn't properly understand things here. I'll just open a pull request with everything I know how to do, and then you can do the rest or help me to do it because it is just a fue lines that i don't know how to do them

**Comment by peter1384** [comment]
@bear101 what to call after ttclient.do for sending a broadcast message? i'm adding broadcast ability to teamtalk android. I'v looked at the qt teamtalk, but it seemze difrent because i didn't understand what to call after ttclient.do in android client

**Reply by bear101** [reply]
all ttclient.do*() commands return a command ID. This command ID can be linked to either CLIENTEVENT_CMD_SUCCESS or CLIENTEVENT_CMD_ERROR event. Then you know if it succeeded or failed

**Comment by peter1384** [comment]
yes, but i don't know what's the name. For example, ttclient.do_broadcast_message userid msg

**Reply by bear101** [reply]
it's doTextMessage()

**Comment by peter1384** [comment]
@bear101 ok, but What's the auder of it,  doTextMessage userid message or doTextMessage message userid?
